### PR TITLE
Extended localnet tests to use the transaction factories, as well

### PR DESCRIPTION
**Pull request by popenta**
Duplicated the tests that run on `localnet` to use the transaction factories. The old tests will be removed in the future, when the functionality will be deleted.

### CODE REVIEW

**Review by CiprianDraghici — DISMISSED** [review]

**Review by CiprianDraghici — APPROVED** [review]

File: src/testutils/contractController.ts [hunk 1]
```diff
@@ -55,4 +58,10 @@ export class ContractController {
         Logger.debug(`ContractController.query [end]: function = ${interaction.getFunction()}, return code = ${bundle.returnCode}`);
         return bundle;
     }
+
+    private getTransactionHash(transaction: Transaction): string {
```
Comment (andreibancioiu) [inline src/testutils/contractController.ts:62]: Generally speaking, `hash` should also have been returned by `await this.provider.sendTransaction(transaction);`.

File: src/transaction.local.net.spec.ts [hunk 2]
```diff
@@ -143,11 +148,14 @@ describe("test transaction", function () {
...
-            getTransaction: async (hash: string) => {
-                return await provider.getTransaction(hash, true);
+        const watcher = new TransactionWatcher(
+            {
+                getTransaction: async (hash: string) => {
+                    return await provider.getTransaction(hash, true);
+                },
             },
-        });
+            { timeoutMilliseconds: 100000 },
```
Comment (andreibancioiu) [inline src/transaction.local.net.spec.ts:157]: Could have added a utility test function such as `createTransactionWatcher`.

**Review by andreibancioiu — APPROVED** [review]

File: src/transaction.local.net.spec.ts (hunk 2 above)
Comment (popenta) [reply src/transaction.local.net.spec.ts:157]: added

File: src/testutils/contractController.ts (hunk 1 above)
Comment (popenta) [reply src/testutils/contractController.ts]: done

**Review by andreibancioiu — APPROVED** [review]

**Review by CiprianDraghici — APPROVED** [review]

**Pull request merged**